tests/benchmarks/peripheral_stress_test/src/timer_temp_wdt_thread.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(timer, LOG_LEVEL_INF);
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/watchdog.h>
+#include "common.h"
+
+static struct k_timer my_timer;
+static int timer_expire_count;
+
+/* Temp sensor */
+static const struct device *temp_dev = DEVICE_DT_GET(DT_NODELABEL(temp_sensor));
+static enum sensor_channel chan_to_use = SENSOR_CHAN_DIE_TEMP;
+
+/* Watchdog */
+extern const struct device *const my_wdt_device;
+extern int my_wdt_channel;
+
+void my_work_handler(struct k_work *work)
+{
+	int rc;
+	struct sensor_value val;
+	int32_t temp_val;
+
+	/* Feed watchdog */
+	rc = wdt_feed(my_wdt_device, my_wdt_channel);
+	if (rc < 0) {
+		LOG_ERR("wdt_feed() returned: %d", rc);
+	}
+
+	/* Capture TEMP sample */
+	rc = sensor_sample_fetch_chan(temp_dev, chan_to_use);
+	if (rc < 0) {
+		LOG_ERR("sensor_sample_fetch_chan() returned: %d", rc);
+	}
+
+	/* Retrieve TEMP sample */
+	rc = sensor_channel_get(temp_dev, chan_to_use, &val);
+	if (rc < 0) {
+		LOG_ERR("sensor_channel_get() returned: %d", rc);
+	}
+	temp_val = (val.val1 * 100) + (val.val2 / 10000);
+
+	timer_expire_count++;
+	LOG_INF("Timer expired %d times, DIE_TEMP is %d.%02u",
+		timer_expire_count, temp_val / 100, abs(temp_val) % 100);
+}
+
+K_WORK_DEFINE(my_work, my_work_handler);
+
+void my_timer_handler(struct k_timer *dummy)
+{
+	k_work_submit(&my_work);
+}
+
+/* Timer thread */
+static void timer_thread(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	atomic_inc(&started_threads);
+
+	if (!device_is_ready(temp_dev)) {
+		LOG_ERR("Device %s is not ready.", temp_dev->name);
+		atomic_inc(&completed_threads);
```

Comment:
Not needed here? missing increment will allow to detect that something has failed

Reply:
Failure detected when LOG_INF("Timer thread has completed, watchdog is stopped"); is missing.